legend-engine-xt-nonrelationalStore-mongodb-executionPlan-test/src/main/resources/core_mongodb_execution_test/mongodb_execution_tests.pure

```diff
@@ -0,0 +1,449 @@
+  testFilterExecution($config, $filterLambda, '/core_mongodb_execution_test/expected_execution_results/combinations/testMultipleAndNestedFilters.json', personTreeFull());
+}
+
+
+// UTILS
+
+function meta::external::store::mongodb::executionTest::executeAndAssert(f: FunctionDefinition<Any>[1], expectedResult: String[1]):Boolean[1]
+ {
+
+ let executionContext = ^meta::external::store::mongodb::functions::pureToDatabaseCommand::MongoDBExecutionContext(queryTimeOutInSeconds=5, enableConstraints=false);
+  
+   let result = meta::legend::execute(
+    $f,
+    [],
+    $executionContext,
+    meta::external::store::mongodb::executionPlan::platformBinding::legendJava::mongoDBLegendJavaPlatformBindingExtensions()
+  )->meta::json::parseJSON()->meta::json::toPrettyJSONString();
+  
+  assertEquals($expectedResult, $result);
+}
+
+
+function meta::external::store::mongodb::executionTest::testCase::utils::testFilterExecution(config:TestConfig[1], filterLambda:LambdaFunction<{Person[1]->Boolean[1]}>[1], expectedResultFileLoc: String[1]):Boolean[1]
+{
+
+testFilterExecution($config, $filterLambda, $expectedResultFileLoc, personTreeFull());
+
+}
+
+function meta::external::store::mongodb::executionTest::testCase::utils::testFilterExecution(config:TestConfig[1], filterLambda:LambdaFunction<{Person[1]->Boolean[1]}>[1], expectedResultFileLoc: String[1], graphFetch: meta::pure::graphFetch::RootGraphFetchTree<Person>[1]):Boolean[1]
+{
+    
+  let mapping = $config.mapping;
+  let runtime = $config.runtime;
+
+  let personQuery = {|Person.all()->from($mapping, $runtime)->filter($filterLambda)
+  ->graphFetch($graphFetch)->serialize($graphFetch)};
+
+  let expectedResult = readFile($expectedResultFileLoc)->toOne();
+  executeAndAssert($personQuery, $expectedResult);
+}
+
+
+function <<paramTest.AfterPackage>> meta::external::store::mongodb::executionTest::testCase::cleanupTestSuite(config:TestConfig[1]):Boolean[1]
+{
+  println('cleanupTestSuite:');
+  println($config.tag);
+  $config.tag->meta::external::store::mongodb::executionTest::utils::stopMongoDBTestServer();
+  true;
+}
+
+function <<access.private>> meta::external::store::mongodb::executionTest::v7::collectTest(tag: String[1]): PureTestCollection[1]
+{
+  println('Collecting tests for MongoDB image tag:');
+  println($tag);
+
+  // in case we failed to stop with above AfterPackage cleanup
+  meta::external::store::mongodb::executionTest::utils::stopMongoDBTestServer($tag);
+
+  let url = meta::external::store::mongodb::executionTest::utils::startMongoDBTestServer($tag);
+
+  let testDataToInsert = meta::pure::functions::io::readFile('/core_mongodb_execution_test/test_setup/person.json')->toOne();
+
+  meta::external::store::mongodb::executionTest::utils::requestMongoDBTestServer($tag, $testDataToInsert);
```

Comment:
is this the new executeInDB() - is this how we want to insert data?  Thought we moved away from that,  If ES is doing it, let us keep it that way, but I thought the test data inserts is part of TestConnection (runtime) which is where you pass in data to insert?  (that is how it works in studio)

Reply:
This does seem like a similar pattern that they had identified as anti-pattern for their test setup but it is used in the latest elastic search store execution test modules, as previously mentioned will send out a mail for us to reach consensus